test/e2e/vsphere_volume_ops_strom.go

```diff
@@ -0,0 +1,188 @@
+		By(fmt.Sprintf("Running test with VOLUME_OPS_SCALE: %v", volume_ops_scale))
+		By("Creating Storage Class")
+		scParameters := make(map[string]string)
+		scParameters["diskformat"] = "thin"
+		storageClassSpec := getVSphereStorageClassSpec("thinsc", scParameters)
+		storageclass, err = client.StorageV1beta1().StorageClasses().Create(storageClassSpec)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating PVCs using the Storage Class")
+		count := 0
+		for count < volume_ops_scale {
+			pvclaimSpec := getVSphereClaimSpecWithStorageClassAnnotation(namespace, storageclass)
+			claim, err := client.CoreV1().PersistentVolumeClaims(namespace).Create(pvclaimSpec)
+			Expect(err).NotTo(HaveOccurred())
+			pvclaims = append(pvclaims, claim)
+			count++
+		}
+
+		By("Waiting for all claims to be in bound phase")
+		for _, claim := range pvclaims {
+			err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, namespace, claim.Name, framework.Poll, framework.ClaimProvisionTimeout)
+			Expect(err).NotTo(HaveOccurred())
+			// Get new copy of the claim
+			claim, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(claim.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Get the bounded PV
+			persistentvolume, err := client.CoreV1().PersistentVolumes().Get(claim.Spec.VolumeName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			persistentvolumes = append(persistentvolumes, persistentvolume)
+		}
+
+		By("Creating pod to attach PV to the node")
+		podSpec := &v1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "pod-30-volumes-",
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "volume-tester",
+						Image:   "gcr.io/google_containers/busybox:1.24",
+						Command: []string{"/bin/sh"},
+						Args:    []string{"-c", "while true ; do sleep 2 ; done"},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
+		var volumeMounts []v1.VolumeMount
+		var volumes []v1.Volume
+		for index, pvclaims := range pvclaims {
+			volumename := fmt.Sprintf("volume%v", index+1)
+			volumeMounts = append(volumeMounts, v1.VolumeMount{Name: volumename, MountPath: "/mnt/" + volumename})
+			volumes = append(volumes, v1.Volume{Name: volumename, VolumeSource: v1.VolumeSource{PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{ClaimName: pvclaims.Name, ReadOnly: false}}})
+		}
+		podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
+		podSpec.Spec.Volumes = volumes
+
+		pod, err := client.CoreV1().Pods(namespace).Create(podSpec)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for pod to be running")
+		Expect(framework.WaitForPodNameRunningInNamespace(client, pod.Name, namespace)).To(Succeed())
+
+		// get fresh pod info
+		pod, err = client.CoreV1().Pods(namespace).Get(pod.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
moved to func
`func createPod(client clientset.Interface, namespace string, pvclaims []*v1.PersistentVolumeClaim) *v1.Pod`